src/test/unit/profile-data.test.js

```diff
@@ -637,6 +643,27 @@ describe('upgrades', function() {
       serializedOldProcessedProfile6
     );
     compareProcessedProfiles(upgradedProfile6, afterUpgradeReference);
+
+    const serializedOldProcessedProfile7 = require('../fixtures/upgrades/processed-7.json');
+    const upgradedProfile7 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7
+    );
+    compareProcessedProfiles(upgradedProfile7, afterUpgradeReference);
+
+    const serializedOldProcessedProfile7a = require('../fixtures/upgrades/processed-7a.json');
+    const afterUpgradeReference8a = unserializeProfileOfArbitraryFormat(
+      require('../fixtures/upgrades/processed-8a.json')
+    );
+    const upgradedProfile7a = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile7a
+    );
+    compareProcessedProfiles(upgradedProfile7a, afterUpgradeReference8a);
+
+    const serializedOldProcessedProfile8 = require('../fixtures/upgrades/processed-8.json');
+    const upgradedProfile8 = unserializeProfileOfArbitraryFormat(
+      serializedOldProcessedProfile8
+    );
+    compareProcessedProfiles(upgradedProfile8, afterUpgradeReference);
```

Comment:
Could you please also add the same "safeguard" kind of test in the following test case ?

Thanks a lot !